x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -748,6 +748,88 @@ public void testOutOfRangeFilterPushdown() {
         }
     }
 
+    public void testOutOfRangeFilterForDoubleFieldsPushdown() {
+        var allTypeMappingAnalyzer = makeAnalyzer("mapping-all-types.json", new EnrichResolution());
+
+        String smallerThanFloat = String.valueOf(randomDoubleBetween(-Double.MAX_VALUE, Float.MIN_VALUE - 1d, true));
+        String largerThanFloat = String.valueOf(randomDoubleBetween(Float.MAX_VALUE + 1d, Double.MAX_VALUE, true));
+
+        List<OutOfRangeTestCase> cases = List.of(
+            new OutOfRangeTestCase("float", smallerThanFloat, largerThanFloat),
+            new OutOfRangeTestCase("half_float", smallerThanFloat, largerThanFloat)
+        );
+
+        final String LT = "<";
+        final String LTE = "<=";
+        final String GT = ">";
+        final String GTE = ">=";
+        final String EQ = "==";
+        final String NEQ = "!=";
+
+        for (OutOfRangeTestCase testCase : cases) {
+            List<String> rangePredicates = List.of(
+                LT + testCase.tooHigh,
+                LTE + testCase.tooHigh,
+                GT + testCase.tooLow,
+                GTE + testCase.tooLow,
+                LT + testCase.tooLow,
+                LTE + testCase.tooLow,
+                GT + testCase.tooHigh,
+                GTE + testCase.tooHigh
+            );
+
+            for (String rangePredicate : rangePredicates) {
+                String comparison = testCase.fieldName + rangePredicate;
+                var query = "from test | where " + comparison;
+
+                Source expectedSource = new Source(1, 18, comparison);
```

Comment:
Declaration could be taken out of each loop.

Reply:
Turns out it's not possible because it depends on `comparison` which is computed inside the loop